reladomographql/src/main/java/com/gs/reladomo/graphql/SchemaProvider.java

```diff
@@ -0,0 +1,111 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URL;
+
+import static graphql.schema.idl.TypeRuntimeWiring.newTypeWiring;
+
+public class SchemaProvider
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger (SchemaProvider.class);
+
+    private SchemaProvider ()
+    {
+    }
+
+    public static GraphQLSchema forResource (String schemaResourceName)
+    {
+        URL url = Resources.getResource (schemaResourceName);
+        LOGGER.info ("Loading from " + url.getPath ());
+        URL metaUrl = Resources.getResource ("meta.graphqls");
+        String metaSdl = null;
+        try
+        {
+            metaSdl = Resources.toString (metaUrl, Charsets.UTF_8);
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException ("Failed to read " + metaUrl, e);
+        }
+
+        try
+        {
+            String sdl = Resources.toString (url, Charsets.UTF_8) + "\n" + metaSdl;
+            GraphQLSchema graphQLSchema = new SchemaProvider ().buildSchema (sdl);
+            GraphQL.newGraphQL (graphQLSchema).build ();
+
+            return graphQLSchema;
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException ("Failed to read config from " + url, e);
+        }
+    }
+
+    /**
+     * see: https://www.graphql-java.com/documentation/v12/schema/
+     */
+    private GraphQLSchema buildSchema (String sdl)
+    {
+        TypeDefinitionRegistry typeRegistry = new graphql.schema.idl.SchemaParser ().parse (sdl);
+        RuntimeWiring runtimeWiring = buildWiring ();
+        SchemaGenerator schemaGenerator = new SchemaGenerator ();
+        return schemaGenerator.makeExecutableSchema (typeRegistry, runtimeWiring);
+    }
+
+    private RuntimeWiring buildWiring ()
+    {
+        RuntimeWiring.Builder builder = RuntimeWiring.newRuntimeWiring ();
+        builder = builder.scalar (ExtendedScalars.DateTime).scalar (ExtendedScalars.Date).scalar (ExtendedScalars.Url).scalar (new TimestampScalar());
+
+        for (MithraRuntimeCacheController each : MithraManagerProvider.getMithraManager ().getRuntimeCacheControllerSet ())
+        {
+            String className = each.getClassName ();
+            String name = StringUtils.decapitalize (className, className.lastIndexOf ('.') + 1);
+            LOGGER.info ("building " + name + " for " + className);
+
+            builder = builder
+                    .type (newTypeWiring ("Mutation")
+                            .dataFetcher (name + "_insert", new ReladomoMutationFetcher (name, each.getClassName (), each.getFinderInstance ())))
+                    .type (newTypeWiring ("Query")
+                            .dataFetcher (name + "ById", new ReladomoQueryFetcher (each.getFinderInstance (), false)))
+                    .type (newTypeWiring ("Query")
+                            .dataFetcher (name + "s", new ReladomoQueryFetcher (each.getFinderInstance (), true)))

```

Comment:
Plural is name+"s"? Like Companys? Consider a name like AllCompany instead, to avoid the problem of English plurals.

Reply:
I possibly can implement better approximation of the English language plural spelling. Would it work? Alternatively should I use "EveryCompany" as a proper prase?

Reply:
EveryCompany sounds good. There's https://github.com/plural4j/plural4j but I don't think it's that robust.

Reply:
I ended up using the same logic as the rest of the Reladomo (support of the "-s", "-es", "-x" and "-y" endings).

I will look at the plural4j at some point, but it depends on the plural count. I need to think about it a bit more.